deposit_contract.sol

```diff
@@ -131,13 +131,14 @@ contract DepositContract is IDepositContract {
     function to_little_endian_64(uint64 value) internal pure returns (bytes memory ret) {
         // Unrolled the loop here.
         ret = new bytes(8);
-        ret[0] = bytes1(uint8(value));
-        ret[1] = bytes1(uint8(value >> 8));
-        ret[2] = bytes1(uint8(value >> 16));
-        ret[3] = bytes1(uint8(value >> 24));
-        ret[4] = bytes1(uint8(value >> 32));
-        ret[5] = bytes1(uint8(value >> 40));
-        ret[6] = bytes1(uint8(value >> 48));
-        ret[7] = bytes1(uint8(value >> 56));
+        bytes8 bytesValue = bytes8(value);
+        ret[0] = bytesValue[0];
```

Comment:
So this works because `uint64` -> `bytes8` casting switches the byteorder and then we are copying it over to `bytes`.

Reply:
Ah sorry, no, the indices have to be reversed!